### ISSUE TITLE
Statistics not shown: translation issue?

### ISSUE THREAD

**Issue by Atrejoe** [allinurl/goaccess#1804]
# Symptom
Statistics are not show in the panel. Graphs look different too and don't show legend nor tickmarks:

![image](https://user-images.githubusercontent.com/585091/83977374-681c4a00-a900-11ea-9f85-6fdaaada846b.png)

# Error messages:
When  inspecting the page using client tools (in Firefox, Chrome and Edge), I get the same error message.

`attribute transform: Expected number, "translate(NaN,0)".`

followed by:
```
report_built_from_source.html:1 Uncaught TypeError: Cannot read property 'substr' of undefined
    at Object.formatDate (report_built_from_source.html:1)
    at Object.fmtValue (report_built_from_source.html:1)
    at SVGTextElement.<anonymous> (report_built_from_source.html:1)
    at SVGTextElement.<anonymous> (report_built_from_source.html:1)
    at report_built_from_source.html:1
    at Y (report_built_from_source.html:1)
    at Array._a.each (report_built_from_source.html:1)
    at Array._a.text (report_built_from_source.html:1)
    at SVGGElement.<anonymous> (report_built_from_source.html:1)
    at report_built_from_source.html:1
```

> note: I open this file as a generated file from disk, not as served from a web server

# GoAccess Command
I run GoAccess from Docker, suing docker-compose.
Using latest docker image:

`/srv/logs/access.log --persist --restore --db-path /srv/data/all/ -o /srv/report/report.html`

My `docker-compose.yml` file:
```docker-compose
version: "3"
services:
  goaccess_all:
    image: allinurl/goaccess:latest
    command: /srv/logs/access.log --persist --restore --db-path /srv/data/all/ -o /srv/report/report.html
    volumes:
      - './data:/srv/data'          #Path to Geoip databases
      - '../NGinx/log:/srv/logs:ro' #Path where NGinx stores logfiles
      - './html:/srv/report'        #Path to output report
      - './conf:/etc/goaccess'      #Path to config
    restart: always
```

My conf file (maybe not relevant).
```
time-format %H:%M:%S
date-format %d/%b/%Y
log-format VCOMBINED
config-dialog false
hl-header true
json-pretty-print false
no-color false
no-column-names false
no-csv-summary false
no-progress false
no-tab-scroll false
with-mouse false
agent-list true
with-output-resolver true
exclude-ip 127.0.0.1
exclude-ip 192.168.0.1-192.168.0.100
exclude-ip MYPUBLICIPHERE
http-method yes
http-protocol yes
no-query-string false
no-term-resolver false
444-as-404 false
4xx-to-unique-count false
all-static-files true
date-spec hr
double-decode false
ignore-crawlers false
crawlers-only false
real-os true
static-file .css
static-file .js
static-file .jpg
static-file .png
static-file .gif
static-file .ico
static-file .jpeg
static-file .pdf
static-file .csv
static-file .mpeg
static-file .mpg
static-file .swf
static-file .woff
static-file .woff2
static-file .xls
static-file .xlsx
static-file .doc
static-file .docx
static-file .ppt
static-file .pptx
static-file .txt
static-file .zip
static-file .ogg
static-file .mp3
static-file .mp4
static-file .exe
static-file .iso
static-file .gz
static-file .rar
static-file .svg
static-file .bmp
static-file .tar
static-file .tgz
static-file .tiff
static-file .tif
static-file .ttf
static-file .flv
geoip-database /srv/data/GeoLite2-City.mmdb
```

**Comment by allinurl** [comment]
Thanks for reporting it! Can I ask you, are you able to replicate this running goaccess without docker? I just want to make sure I can pinpoint this better from my end. Thank you!

**Comment by Atrejoe** [comment]
My apologies for not responding sooner. 
How do you suggest I test this, by just installing v1.4 like stated here : https://goaccess.io/download ?
As this appears to be a client-side issue, is there a way we can have a unminimized html file for debugging purposes?

**Comment by Atrejoe** [comment]
Using WSL2 and Ubuntu, I was able to get a true Linux experience, installing goaccess (1.4) via a simple ` apt-get` .

This allowed me to do a few fast iterations; it appears the issues occurs when creating reports using persistent data.
Using a copy of the persisted data and the configuration file as mentioned above:

Produces fine report (but is not complete)
`goaccess -p goaccess.conf access_cyclingsouth.log -o report_cyclingsouth.html`

Produces problematic report
` goaccess -p goaccess.conf access_cyclingsouth.log -o report_cyclingsout_persisted.html --persist --restore --db-path cyclingsouth`

Compared visually:
![image](https://user-images.githubusercontent.com/585091/95686352-8fac5780-0bfd-11eb-8806-4061a74d8fdb.png)

**Comment by allinurl** [comment]
Thanks for the additional details. Can I ask you to please build from [**development**](https://github.com/allinurl/goaccess#build-from-github-development), I pushed a fix that address some issues when persisting/restoring data from disk. Though, you may need to delete the old database files and parse the log from scratch so it stores the right data. 

Let me know how it goes. Thank you!

**Comment by Atrejoe** [comment]
I didn't build from source previously as I couldn't get it to work straight away.
After installing *all* the prerequisites, I was able to build (from master). (hurray!)

However, the result stayed the same....without deleting the database.

Then I read your comment regarding deletion of the database, after which a database rebuild would occur.
Database corruption leading to strange visual output is a bit alarming, but ok, I gave it a try.

After deleting the database, all looked well again...great!

Then, and this may be a different subject, although my logs contain data from about February this year, the report only shows metrics from more or less the last month.
When rebuilding with persisted data and newer logs, all still looks fine, but the reported period has shifted.
This may be normal behavior for GoAccess: only report the last x days, even if the database contains more data.

Screenshots:
- 1st import
- 2nd import (using `--persist` and `--restore`)
- Start of logfile being parsed

![image](https://user-images.githubusercontent.com/585091/95784179-a1602e80-0cd3-11eb-8bdb-c31727b888f7.png)

Note: all aggregate metrics appear to have doubled after reimporting.
Am I missing something? Is there a way to view the content of the databases (in a comfortable way), and if so what files are relevant?

**Comment by allinurl** [comment]
@Atrejoe Thanks for the update.

Regarding your points:

- The numbers reported are from Feb. However, the report only shows 365 entries, in your case, it's maxing out since you are reporting the at the hour level. I'm not sure if showing 365 days would serve the purpose when displaying at the hour level. Also, not sure if folks will go over the possible thousands of entries .Thoughts? 

- Like you said, metrics appear to have doubled. The patch fixed an issue where it would not account those when restoring. The easiest would be to run grep over all of your logs from `Feb` for specific request and see if the count matches the output of goaccess. e.g., 

      zgrep 'zwagger/index.html' access.log* | wc -l

**Comment by allinurl** [comment]
For accuracy, should be:

    zgrep 'GET /zwagger/index.html HTTP/1.1' access.log* | wc -l

**Comment by Atrejoe** [comment]
Given that my problem was resolved by deleting the database files, I deleted them on my docker instance too.
I also changed the porting resolution to `date` and enlarged to 732 segments (roughly two years).

I then started the containers and all was well!

I don't know the cause of the problem still, but at least (for me) this is not a problem anymore (for now).

If you want me to do more logging on why this was failing in the first place, let me know.

**Comment by Atrejoe** [comment]
@allinurl I did not yet, I am bit swamped, about to not do it :)

but here you go:
![image](https://user-images.githubusercontent.com/585091/95965776-9cea6180-0e0a-11eb-803f-74506e22ae60.png)

I think the Html may under-report because it filters out my ip address. (you were checking for over-reporting anyway, right?)

**Comment by allinurl** [comment]
@Atrejoe That's right. Like you said, if you are filtering out your IP, that could explain the reason. Thank you for posting that!
